### PR TITLE
added login and reg

### PR DESCRIPTION
**Pull request by xerxes-on**

### CODE REVIEW

File: src/api/client.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import axios from "axios";
+// import { useAuthStore } from "@/stores/auth";
+
+// const authStore = useAuthStore();
+
+const client =  axios.create({
+        baseURL: 'http://bookrating.test/api/'
```
Comment (volhaliferovich) [inline src/api/client.js:7]: Remove "/" at the end of your base url if you want to use "/books"

File: src/components/common/Svg.vue [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<script setup>
+import {defineProps, ref} from "vue";
```
Comment (volhaliferovich) [inline src/components/common/Svg.vue:2]: You don't need to import defineProps, it's a compiler macro. Remove it from here

File: src/plugins/index.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { createPinia } from 'pinia'
+import router from "../router";
+import {ref} from "vue";
+const loading = ref(false)
```
Comment (volhaliferovich) [inline src/plugins/index.js:4]: Losing shouldn't be here, if you want to create a global loading variable, create a new store under the store folder and initialise it there

File: src/plugins/index.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+export function registerPlugins(app) {
+    app.provide('loading', loading);
```
Comment (volhaliferovich) [inline src/plugins/index.js:6]: remove this

File: src/api/client.js (hunk 1 above)
Comment (xerxes-on) [reply src/api/client.js:7]: ok

File: src/components/common/Svg.vue (hunk 2 above)
Comment (xerxes-on) [reply src/components/common/Svg.vue:2]: ok

File: src/plugins/index.js (hunk 4 above)
Comment (xerxes-on) [reply src/plugins/index.js:6]: removed

File: src/App.vue [hunk 5]
```diff
@@ -1,9 +1,22 @@
 <script setup>
-import { RouterLink, RouterView } from 'vue-router'
+import {RouterView} from 'vue-router'
```
Comment (volhaliferovich) [inline src/App.vue:2]: You don't need to import RouterView, it's available without it

File: src/router/index.js [hunk 6]
```diff
@@ -1,21 +1,37 @@
...
+            name: 'home',
+            component: () => import('@/views/Home.vue'),
+        },
+    ],
 })
+router.beforeEach((to, from, next) => {
+    const authStore = useAuthStore();
+    const isAuthenticated = authStore.isAuthorized;
+
+    if (!isAuthenticated && (to.name !== 'login' ||) ) {
```
Comment (volhaliferovich) [inline src/router/index.js:28]: You haven't finished the condition here, please fix, even if you already did it in a different branch

File: .prettierrc.json [hunk 7]
```diff
@@ -0,0 +1,5 @@
+{
+    "tabWidth": 4,
+    "semi": false,
+    "singleQuote": true
```
Comment (volhaliferovich) [inline .prettierrc.json:4]: Please add these rules:  "trailingComma": "es5", "printWidth": 180. And run "npm run format" again.

File: src/components/loginComponents/LoginForm.vue [hunk 8]
```diff
@@ -1,36 +1,96 @@
 <script setup>
+import authAPI from '@/api/auth.js'
+import Svg from '@/components/common/Svg.vue'
+import { useVuelidate } from '@vuelidate/core'
+import { required, email } from '@vuelidate/validators'
+import { reactive, ref } from 'vue'
+import { useAuthStore } from '@/stores/auth.js'
+import { useRouter } from 'vue-router'
+import { useErrorPop } from '@/stores/index.js'
```
Comment (volhaliferovich) [inline src/components/loginComponents/LoginForm.vue:9]: Always remove unused imports.

File: src/components/loginComponents/RegisterForm.vue [hunk 9]
```diff
@@ -1,78 +1,179 @@
...
-import Svg from "@/components/Svg.vue";
-import {ref} from "vue";
-const showPassword = ref(false);
+import Svg from '@/components/common/Svg.vue'
+import { reactive, ref, toRef } from 'vue'
+import { email, required, sameAs } from '@vuelidate/validators'
+import { useVuelidate } from '@vuelidate/core'
+import authAPI from '@/api/auth.js'
+import { useAuthStore } from '@/stores/auth.js'
+import { useRouter } from 'vue-router'
```
Comment (volhaliferovich) [inline src/components/loginComponents/RegisterForm.vue:8]: Change the order of imports everywhere. You should sort them from vue imports to your own custom ones (vue, vue-router, vuelidate, your stores/api, your components)

File: src/views/Register.vue [hunk 10]
```diff
@@ -0,0 +1,40 @@
+<script setup>
+import LoginForm from '@/components/loginComponents/LoginForm.vue'
+import Svg from '@/components/common/Svg.vue'
+import RegisterForm from '@/components/loginComponents/RegisterForm.vue'
+import Image from '@/components/common/Image.vue'
+import Logo from '@/components/common/Logo.vue'
```
Comment (volhaliferovich) [inline src/views/Register.vue:6]: Unused imports, LoginForm as well.

File: index.html [hunk 11]
```diff
@@ -1,16 +1,21 @@
...
+        <link rel="icon" href="/favicon.ico" />
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+        <title>Vite App</title>
+        <link rel="stylesheet" href="src/output.css" />
+        <!--      fontawesome-->
+        <script
+            src="https://kit.fontawesome.com/9c18f8edbf.js"
+            crossorigin="anonymous"
+        ></script>
+        <script src="https://cdn.tailwindcss.com"></script>
```
Comment (volhaliferovich) [inline index.html:14]: You already have tailwindcss in your package.json. You should go either with npm installations, or cdn.

**Review by volhaliferovich — CHANGES_REQUESTED** [review]

File: src/api/client.js [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import axios from 'axios'
+// import { useAuthStore } from "@/stores/auth";
+
+// const authStore = useAuthStore();
```
[the remark this answers is not in the document]
Comment (xerxes-on) [reply src/api/client.js:4]: okay

File: .prettierrc.json (hunk 7 above)
Comment (xerxes-on) [reply .prettierrc.json:4]: done

**Review by volhaliferovich — APPROVED** [review]

**Pull request merged**